main.py

```diff
@@ -30,46 +30,93 @@ def hello():
-@app.route('/circles/api/v1.0/users/<int:person_id>', methods=['GET'])
-def get_person(person_id):
+@app.route('/circles/api/v1.0/users/<int:person_id>/', defaults={'resource': None})
+@app.route('/circles/api/v1.0/users/<int:person_id>/<resource>', methods=['GET'])
+def get_person(person_id, resource):
     # Request all users.
     if not person_id:
         return jsonify(Person=get_all_nodes(Person))
-
-    # Request specific user.
+    # Fetch the person
     person = Person.match(graph, person_id).first()
     if not person:
         abort(404, description="Resource not found")
-    return jsonify(Person=person.json_repr())
-
-
-@app.route('/circles/api/v1.0/circles/', defaults={'circle_id': None})
-@app.route('/circles/api/v1.0/circles/<int:circle_id>', methods=['GET'])
-def get_circle(circle_id):
+    if not resource:
+        # Request specific user.
+        return jsonify(person.json_repr())
+
+    # Request specific resource associated with the person
+    output = []
+    if resource not in ['circles', 'events', 'knows']:
+        abort(404, description="Invalid resource specified")
+
+    if resource == 'circles':
+        for c in person.IsMember:
+            output.append(c.json_repr())
+    if resource == 'events':
+        for e in person.InvitedTo:
+            output.append(e.json_repr())
+    if resource == 'knows':
+        for k in person.Knows:
+            output.append(k.json_repr())
+    return jsonify(output)
+
+
+@app.route('/circles/api/v1.0/circles/<int:circle_id>/', defaults={'resource': None})
+@app.route('/circles/api/v1.0/circles/<int:circle_id>/<resource>', methods=['GET'])
+def get_circle(circle_id, resource):
     # Request all circles.
     if not circle_id:
         return jsonify(Circle=get_all_nodes(Circle))
 
-    # Request specific circle.
+    # Fetch circle.
     circle = Circle.match(graph, circle_id).first()
     if not circle:
         abort(404, description="Resource not found")
-    return jsonify(Circle=circle.json_repr())
-
-
-@app.route('/circles/api/v1.0/events/', defaults={'event_id': None})
-@app.route('/circles/api/v1.0/events/<int:event_id>', methods=['GET'])
-def get_event(event_id):
+    if not resource:
+        # Request specific circle
+        return jsonify(circle.json_repr())
+
+    # Request specific resource associated with the circle
+    output = []
+    if resource not in ['members', 'events']:
+        abort(404, description="Invalid resource specified")
+
+    if resource == 'members':
+        for m in circle.HasMember:
+            output.append(m.json_repr())
+    if resource == 'events':
+        for e in circle.Scheduled:
+            output.append(e.json_repr())
+    return jsonify(output)
+
+
+@app.route('/circles/api/v1.0/events/<int:event_id>/', defaults={'resource': None})
+@app.route('/circles/api/v1.0/events/<int:event_id>/<resource>', methods=['GET'])
+def get_event(event_id, resource):
     # Request all events.
     if not event_id:
```

Comment:
I don't think `event_id` can be None anymore, so we can remove this section.

Reply:
Can't remove this-- the optional value is the resource being requested

absence of a resource will give the object itself (/jwei/) versus the presence of a resource will give something that is an edge away (/jwei/circles)